extension/dynamicconfig/model/config.go

```diff
@@ -0,0 +1,81 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"fmt"
+	"strings"
+
+	com "github.com/open-telemetry/opentelemetry-proto/gen/go/common/v1"
+	res "github.com/open-telemetry/opentelemetry-proto/gen/go/resource/v1"
+)
+
+type Config struct {
+	ConfigBlocks []*ConfigBlock // TODO: ok with viper? then convert others, too
+}
+
+func (config *Config) Match(resource *res.Resource) *ConfigBlock {
+	resourceSet, resourceList := embed(resource)
+	totalBlock := &ConfigBlock{
+		Resource:     resourceList,
+		MetricConfig: &MetricConfig{},
+	}
+
+	for _, block := range config.ConfigBlocks {
+		if doInclude(block, resourceSet) {
+			totalBlock.Add(block)
+		}
+	}
+
+	return totalBlock
+}
+
+func embed(resource *res.Resource) (map[string]bool, []string) {
+	resourceSet := make(map[string]bool)
+	resourceList := make([]string, len(resource.Attributes))
+
+	for i, attr := range resource.Attributes {
+		attrString := attrToString(attr)
+		resourceSet[attrString] = true
+		resourceList[i] = attrString
+	}
+
+	return resourceSet, resourceList
+}
+
+func attrToString(attr *com.KeyValue) string {
+	rawValue := attr.Value.String()
+	value := strings.Split(rawValue, ":")[1]
+	attrString := clean(fmt.Sprintf("%s:%s", attr.Key, value))
+
+	return attrString
+}
+
+func doInclude(block *ConfigBlock, resourceSet map[string]bool) bool {
+	include := true
```

Comment:
So if a config block resource does not have any label, then it will match any resource?

Is this working as intended? Let's add a comment and test regardless this is desired or not.

Reply:
Yup, that's correct, this is intended behavior. The idea was to include a mechanism for being able to specify default configs that could apply to all resources. We could use something like the `*` for Patterns, but it seemed more natural for the user to specify no resource. Thus, if a user specifies 1 or more resources, the configs would be keyed specifically to a particular resource. Otherwise, the configs apply to all resources.

Adding documentation to this effect. Also, the tests feature a config block with no resource specified, which corresponds to this case.